### PR TITLE
Moves Clown/Mime/Borg/Relgion nameing to prefs

### PR DESCRIPTION
**Pull request by Aurorablade**
What do?
Most of this is ported from a varient of TG that was close to our current prefrence window set up.
You can now set you special names in the pref window

This does change some functionality on renaming escpially for borgs.
Namepick verb now resets it to your pref set name.
For syndicate the modify name verb is adjusted to work on ALL syndiborg modules. Every module but Sabotour can select a new name once (or namepick and go with your pref). Sabs however should still be able to multipick.
Chaplains must use the bible one time to change the icons.

:cl: Fethas
tweak:Chaplains/Mimes/Borgs/AI now set names via the pref screen.
tweak: Borg name pick will allow borgs created by robotcists to pick a custom name or leave the input blank for a name in prefrences.
tweak: Chaplains use your book to set the icon. Do not hit yourself.
/:cl:

### CODE REVIEW

File: code/game/jobs/job/support_chaplain.dm [hunk 1]
```diff
@@ -36,134 +36,41 @@
 	if(H.mind)
 		H.mind.isholy = TRUE
 
-	spawn()
```
Comment (Fox-McCloud) [inline code/game/jobs/job/support_chaplain.dm]: die in a fire.

File: code/game/objects/items/weapons/storage/bible.dm [hunk 2]
```diff
@@ -1,3 +1,13 @@
+
+//Pretty bible names
+var/global/list/biblenames =		list("Bible", "Koran", "Scrapbook", "Creeper", "White Bible", "Holy Light", "PlainRed", "Tome", "The King in Yellow", "Ithaqua", "Scientology", "the bible melts", "Necronomicon", "Greentext")
+
+//Bible iconstates
+var/global/list/biblestates =		list("bible", "koran", "scrapbook", "creeper", "white", "holylight", "athiest", "tome", "kingyellow", "ithaqua", "scientology", "melted", "necronomicon", "greentext")
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/bible.dm]: the heck is with this spacing?

**Comment by Fox-McCloud** [comment]
The best religion and deity name.

File: code/game/objects/items/weapons/storage/bible.dm (hunk 2 above)
Comment (Aurorablade) [reply code/game/objects/items/weapons/storage/bible.dm]: I think they were trying to line it up on TG so you could see what was what. For us it half works.

Comment (Aurorablade) [reply code/game/objects/items/weapons/storage/bible.dm]: Then again TG also has relgion on a SS now...just wasn't going there wiht this PR

Comment (Fox-McCloud) [reply code/game/objects/items/weapons/storage/bible.dm]: These all need to be made proper global lists with `GLOBAL_LIST_INIT`

File: code/game/objects/items/weapons/storage/bible.dm [hunk 3]
```diff
@@ -27,6 +37,68 @@
...
+
+/obj/item/storage/bible/attack_self(mob/living/carbon/human/H)
+	if(!istype(H))
+		return
+	if(ticker && !ticker.Bible_icon_state && H.mind.assigned_role == "Chaplain")
+		//Open bible selection
+		var/dat = "<html><head><title>Pick Bible Style</title></head><body><center><h2>Pick a bible style</h2></center><table>"
+
+		var/i
+		for(i = 1, i < biblestates.len, i++)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/bible.dm]: `for(var/i in 1 to GLOB.biblestates.len)`

File: code/game/objects/items/weapons/storage/bible.dm [hunk 4]
```diff
@@ -27,6 +37,68 @@
...
+			var/icon/bibleicon = icon('icons/obj/storage.dmi', biblestates[i])
+			var/nicename = biblenames[i]
+			H << browse_rsc(bibleicon, nicename)
+			dat += {"<tr><td><img src="[nicename]"></td><td><a href="?src=\ref[src];seticon=[i]">[nicename]</a></td></tr>"}
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/storage/bible.dm]: pretty sure the \ref will throw a runtime. Use `UID`

**Comment by AffectedArc07** [comment]
**This is gonna need config/savefile changes, send help**

We don’t use savefiles, we use the DB to store everything. Probably new columns for `AI name, Borg name, clown name, mime name, religion, deity, (bible style?)`

I can help with the SQL stuff if you need

File: code/modules/client/preference/preferences.dm [hunk 5]
```diff
@@ -200,12 +200,20 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 	var/list/gear = list()
 	var/gear_tab = "General"
 
+	var/list/custom_names = list("human", "clown", "mime", "ai", "cyborg", "religion", "deity")
```
Comment (Fox-McCloud) [inline code/modules/client/preference/preferences.dm]: don't forget the human name. These are "backup humans" for like...events and crap.

File: code/game/objects/items/weapons/storage/bible.dm (hunk 3 above)
Comment (Aurorablade) [reply code/game/objects/items/weapons/storage/bible.dm]: OH GAH GLOBS...I FORGOT...

File: code/modules/client/preference/preferences.dm (hunk 5 above)
Comment (Aurorablade) [reply code/modules/client/preference/preferences.dm]: See i forgot to remove that...Do we even need that?

**Comment by Aurorablade** [comment]
I know affected its in my to-dos.

its also like..330 am and i am going to bed and will address comments after work tomarrow.

File: code/modules/client/preference/preferences.dm (hunk 5 above)
Comment (Fox-McCloud) [reply code/modules/client/preference/preferences.dm]: we'll need it once lavaland goes in, so might as well put it in now.

**Comment by Quantum-M** [comment]
What about late round AI and Cyborg names? I feel those should get the name pick prompt since round start cyborgs/AIs usually have no direct correlation (IC wise) to characters that end up borged or turned into a AI.

**Comment by KasparoVy** [comment]
@AffectedArc07 
yro'ue crop

**Comment by Kyep** [comment]
I support this.

**Comment by Fox-McCloud** [comment]
Were you just trying to be funny with your clown/mime names? 

Only reason I ask is because if you werne't, well, your lists are mixed up.

**Comment by Aurorablade** [comment]
@Fox-McCloud i was wondering who would notice first...

**Comment by Aurorablade** [comment]
note to self work on SQL for this later

**Comment by Aurorablade** [comment]
just got to get the sql update file wrote up later.

**Comment by Aurorablade** [comment]
again no idea about the SQL bit so...back checks.

**Comment by AffectedArc07** [comment]
@Aurorablade want a hand?

**Comment by Aurorablade** [comment]
I mean if you see something wrong feel free to correct!

**Comment by AffectedArc07** [comment]
Update `config/example/dbconfig.txt` to change `SQL_VERSION` from `5` to `6`

**Comment by Aurorablade** [comment]
I don't run an active DB so....why would it be doing that? What did i mess up in the sql setup that i am not seeing..and on the size..i need a suggestion or else i always mess up a ui size.

**Comment by AffectedArc07** [comment]
I would say tack on an extra 100 pixels to the vertical, and I can give you an account on my SQL server if you want

File: code/__DEFINES/misc.dm [hunk 6]
```diff
@@ -315,7 +315,7 @@
 #define INVESTIGATE_BOMB "bombs"
 
 // The SQL version required by this version of the code
-#define SQL_VERSION 7
+#define SQL_VERSION 8
```
Comment (farie82) [inline code/__DEFINES/misc.dm]: should be 9 by now

File: SQL/updates/7-8.sql [hunk 7]
```diff
@@ -0,0 +1,9 @@
+#Updating the SQL from version 7 to version 8. -fethas
```
Comment (farie82) [inline SQL/updates/7-8.sql]: Whole file should be renamed to 8-9.sql

**Comment by Aurorablade** [comment]
yes i know i got to get to it when i can.

**Comment by Fox-McCloud** [comment]
Is any portion of this not working? Any bugs/errors? I know greentext had issues--is that resolved?

**Comment by AffectedArc07** [comment]
image
Aside from this I dont think this had issues

**Comment by Aurorablade** [comment]
YEah and i got to actually setup a local DB to check whats doing it cuase i am not seeing it right off and i keep getting sidetracked actually setting up a local db on my computer on my days off. but i really...REALLLY REALLY REALLY *looks at clock* REALLy got to head to work now.

**Comment by KasparoVy** [comment]
Yeah it's the origin story of the tide, join without futzing with the character creator not one click and let the bwoinks fall where they may

Bald black-eyed 30 y.o. space men are an icon

**Comment by Aurorablade** [comment]
well right now i can't figur out why its not reloading the data back into the fields...i can't see what i am missing but i didn't work on it today due to housework and just taking a brake. Kinda gave myself eyestrain on it the other day.

**Comment by Aurorablade** [comment]
IT FUCKING WORKS NOW

**Comment by KasparoVy** [comment]
MADNESS

**Comment by Fox-McCloud** [comment]
@Aurorablade was the probnlem with greentext ever solved?

If so, I can start reviewing this.

**Comment by Aurorablade** [comment]
By greentext you mean objectives I would need to know a good way to force that to be sure

**Comment by Fox-McCloud** [comment]
No, the greentext Bible. You said it wasn't working:

>Figure out why greentext is not showing up in the bible list...My lists should be matching....WOT

Need to know if this was fixed.

**Comment by Aurorablade** [comment]
yeah updated OP frgot i had that todo list there. It works. just double checked. now off to work.

**Comment by Aurorablade** [comment]
This should be good to go btw.

**Comment by Drakeven** [comment]
This looks neat!

**Comment by Fox-McCloud** [comment]
after the tool refactor, this should take next highest priority.

**Comment by Aurorablade** [comment]
Hold on kicking travis in the ass.

**Comment by Aurorablade** [comment]
DAADDDDDDD TRAVIS IS BULLYING MEEEEE

**Comment by AffectedArc07** [comment]
Sidenote
Make sure to update the DB version in `config/example/dbconfig.txt`

File: code/game/jobs/job/support_chaplain.dm [hunk 8]
```diff
@@ -36,134 +36,41 @@
...
+	var/obj/item/storage/bible/B = new /obj/item/storage/bible(H)
+	H.equip_to_slot_or_del(B, slot_l_hand)
+
+	var/new_religion = "Christianity"
+	if(H.client && H.client.prefs.custom_names["religion"])
+		new_religion = H.client.prefs.custom_names["religion"]
+
+	switch(lowertext(new_religion))
+		if("christianity")
+			B.name = pick("The Holy Bible")
```
Comment (farie82) [inline code/game/jobs/job/support_chaplain.dm]: Don't need the pick here

File: code/modules/mob/living/silicon/robot/syndicate.dm [hunk 9]
```diff
@@ -83,7 +83,21 @@
...
-	rename_self(braintype, TRUE, TRUE)
+
+	var/raw_name = input(usr, "Choose your cyborg name (Leave empty to use default naming scheme):", "Character Preference")  as text|null
+	var/new_synborg_name
+	if(!raw_name)
+		new_synborg_name = get_default_name(designation)
+	else
+		new_synborg_name = reject_bad_name(raw_name,1 )
+	if(new_synborg_name)
+		rename_character(, new_synborg_name)
```
Comment (farie82) [inline code/modules/mob/living/silicon/robot/syndicate.dm]: Missing the first argument here. Prob meant to be null

File: code/modules/mob/mob_helpers.dm [hunk 10]
```diff
@@ -576,37 +576,44 @@ var/list/intents = list(INTENT_HELP,INTENT_DISARM,INTENT_GRAB,INTENT_HARM)
...
-		rename_character(oldname, newname)
+		for(var/mob/living/M in GLOB.player_list)
+			if(M == src)
+				continue
+			if(!newname || M.real_name == newname)
+				newname = null
+				loop++ // name is already taken so we roll again
+				break
+		loop--
+		safety++
```
Comment (farie82) [inline code/modules/mob/mob_helpers.dm]: The whole while is a bit hard to read. Maybe use a do while loop and use the variable `loop` as a logical var (no ++ -- etc)

**Comment by Aurorablade** [comment]
have to get some of thiswhen my computer is back for repair..i mean i could copy a repo onto my laptop but

**Comment by Aurorablade** [comment]
gonna deconflict this tommarow but don't have a sql setup right now so...

**Comment by Aurorablade** [comment]
sorry working retail during covid is crazy i also gto to get access to an SQL setup again to make sure i can test right.

**Comment by Fox-McCloud** [comment]
Any idea if/when this might get updated?

**Comment by Aurorablade** [comment]
i am gonna need to remake the PR most likely due to the SQL changes. I am sorry just i blame retail. I am gonna hard set myself for either weds or thursday to start redoing the PR so if you want to close it in the meantime feel free i can still pull from the branch/closed PR to redo the changes. I may need help on SQL testing will be the only point.

**Comment by AffectedArc07** [comment]
> i am gonna need to remake the PR most likely due to the SQL changes. I am sorry just i blame retail. I am gonna hard set myself for either weds or thursday to start redoing the PR so if you want to close it in the meantime feel free i can still pull from the branch/closed PR to redo the changes. I may need help on SQL testing will be the only point.

**Pull request closed without merging**